pkg/controllers/dynakube/otelc/statefulset/env_test.go

```diff
@@ -106,3 +109,197 @@ func TestEnvironmentVariables(t *testing.T) {
+			proxy:           nil,
+			expectedNoProxy: "",
+		},
+		{
+			name:            "telemetryIngest, public AG, with proxy secret",
+			extensions:      nil,
+			telemetryIngest: &telemetryingest.Spec{},
+			activeGate:      nil,
+			proxy: &value.Source{
+				ValueFrom: testProxySecretName,
+			},
+			expectedNoProxy: "",
+		},
+		{
+			name:            "telemetryIngest, public AG, with proxy value",
+			extensions:      nil,
+			telemetryIngest: &telemetryingest.Spec{},
+			activeGate:      nil,
+			proxy: &value.Source{
+				Value: testProxyValue,
+			},
+			expectedNoProxy: "",
+		},
+		{
+			name:            "telemetryIngest, local AG, without proxy",
+			extensions:      nil,
+			telemetryIngest: &telemetryingest.Spec{},
+			activeGate:      &testActiveGate,
+			proxy:           nil,
+			expectedNoProxy: "",
+		},
+		{
+			name:            "telemetryIngest, local AG, with proxy secret",
+			extensions:      nil,
+			telemetryIngest: &telemetryingest.Spec{},
+			activeGate:      &testActiveGate,
+			proxy: &value.Source{
+				ValueFrom: testProxySecretName,
+			},
+			expectedNoProxy: "dynakube-activegate.dynatrace",
+		},
+		{
+			name:            "telemetryIngest, local AG, with proxy value",
+			extensions:      nil,
+			telemetryIngest: &telemetryingest.Spec{},
+			activeGate:      &testActiveGate,
+			proxy: &value.Source{
+				Value: testProxyValue,
+			},
+			expectedNoProxy: "dynakube-activegate.dynatrace",
+		},
+		{
+			name:            "telemetryIngest, extensions, local AG, without proxy",
+			extensions:      &dynakube.ExtensionsSpec{},
+			telemetryIngest: &telemetryingest.Spec{},
+			activeGate:      nil,
+			proxy:           nil,
+			expectedNoProxy: "",
+		},
+		{
+			name:            "telemetryIngest, extensions, local AG, with proxy secret",
+			extensions:      &dynakube.ExtensionsSpec{},
+			telemetryIngest: &telemetryingest.Spec{},
+			activeGate:      nil,
+			proxy: &value.Source{
+				ValueFrom: testProxySecretName,
+			},
+			expectedNoProxy: "dynakube-extensions-controller.dynatrace,dynakube-activegate.dynatrace",
+		},
+		{
+			name:            "telemetryIngest, extensions, local AG, with proxy value",
+			extensions:      &dynakube.ExtensionsSpec{},
+			telemetryIngest: &telemetryingest.Spec{},
+			activeGate:      nil,
+			proxy: &value.Source{
+				Value: testProxyValue,
+			},
+			expectedNoProxy: "dynakube-extensions-controller.dynatrace,dynakube-activegate.dynatrace",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dk := getTestDynakube()
+			dk.Spec.Extensions = tt.extensions
+			dk.Spec.TelemetryIngest = tt.telemetryIngest
+			dk.Spec.Proxy = tt.proxy
+
+			if tt.activeGate != nil {
+				dk.Spec.ActiveGate = *tt.activeGate
+			}
+
+			dataIngestToken := getTokens(dk.Name, dk.Namespace)
+			configMap := getConfigConfigMap(dk.Name, dk.Namespace)
+			statefulSet := getStatefulset(t, dk, &dataIngestToken, &configMap)
+
+			switch {
+			case tt.proxy == nil:
```

Comment:
i'm big fun of data driven test, but in this case IMO better to go to separate tests, cause it's lot's of logic inside test :)

Reply:
Split tests up in [7bc2317](https://github.com/Dynatrace/dynatrace-operator/pull/4630/commits/7bc23172d110807e4fb55fd5a450a88de1bae893)